go/libraries/doltcore/sqle/dfunctions/dolt_merge.go

```diff
@@ -175,7 +180,14 @@ func mergeIntoWorkingSet(ctx *sql.Context, sess *dsess.DoltSession, roots doltdb
 					return ws, hasConflicts, wsErr
 				}
 
-				ctx.Warn(DoltMergeWarningCode, err.Error())
+				// Give a better error message for unresolved conflicts when autocommit is on. This is because the transaction
+				// gets aborted when merge conflicts arise during autocommit and the dolt_conflicts table is not actually
+				// written to.
+				if autocommit {
```

Comment:
Don't write different error messages for autocommit or not. Giving more details error messages is fine.